Eli/sender mismatch exceptions

biblicabeebli opened the pull request:
(this is the fourth in a sequence of pull requests to get bug fixes and refactors into main)
(contains some benign developer quality of life changes)

Contains the set of fixes needed to correctly handle unhandled errors in push notifications, correctly handling the thirdpartautherror case.

Contains a new settings flag to disable push notification quota errors.

misc:
contains an update to the database dumping alias that I put together because I was dealing with rebuilding my old development environment.

prior pull request
https://github.com/onnela-lab/beiwe-backend/pull/233

biblicabeebli commented:
@zagorsky I would like to get this merged into main because it contains a fix in the handling of failed push notifications.  I don't think there will be any further improvements to push notifications at this level, but this will at least provide correct history for failed pushes.

zagorsky reviewed (changes requested):
@biblicabeebli I have two requests on this:
- [ ] Are the environment variables like `STORE_DECRYPTION_KEY_ERRORS` documented somewhere, like in the README or in an example .env or config file?- [ - [ ] I think the method of evaluating boolean environment variables will return True in situations where you don't want it to (see my inline comment).

biblicabeebli commented on config/settings.py:
```diff
@@ -1,38 +1,39 @@
-STORE_DECRYPTION_KEY_ERRORS = getenv("STORE_DECRYPTION_KEY_ERRORS") or False
+REPORT_DECRYPTION_KEY_ERRORS = bool(getenv("REPORT_DECRYPTION_KEY_ERRORS", False))
+STORE_DECRYPTION_KEY_ERRORS = bool(getenv("STORE_DECRYPTION_KEY_ERRORS", False))
 
```
Fair point, and actually it looks like I've screwed this up on IS_STAGING too.

I'm going to go with this slight modification of the recommendation because I find case-sensitive parameters obnoxious:
`os.getenv('DEBUG', 'false').lower() == 'true'`

I'm sticking IS_STAGING on my todo list, the biggest side effect is that all servers are storing decryption key errors, but I need to review it.  I should probably review parameter read-in general, I've been crappy at it before.

biblicabeebli replied:
I am actually going to entirely remove this staging variable.  There are only 2 uses of IS_STAGING, line error recording and an endpoint to determine if there server is a staging server.  Adding a STORE_DECRYPTION_LINE_ERRORS, removing that endpoint because I needed it exactly once in something like 2015, removing IS_STAGING.  Change in upcoming commit.

biblicabeebli commented:
- Are the environment variables like STORE_DECRYPTION_KEY_ERRORS documented somewhere, like in the README or in an example .env or config file?

No, looks like there isn't, and I don't think there is documentation of several optional parameters either.

I think the best way to do that is to stick a link in the wiki that points to the settings file.  The settings file is readable-enough, and should fully document runtime variables and their effects.  On this branch I went through to ensure that was the case.  I'm going to do so again, extend it further, double check that side effects in the config __init__.py file are reasonable and/or addressed.

biblicabeebli commented:
Looks like we have other unused parameters, E500_EMAIL_ADDRESS, OTHER_EMAIL_ADDRESS, DEFAULT_S3_RETRIES.  I've removed those including in the readme.

I've removed references to the config/constants.py file in the readme, as that now (correctly and finally!) no longer interrogates environment variables.  If something can have its value changed from a default value then it should go in settings.py.

This means I've removed references to these variables from the readme, they are now documented in setttings.py:
CONCURRENT_NETWORK_OPS
FILE_PROCESS_PAGE_SIZE

And these are no longer referenced (and IMO shouldn't be referenced):
ASYMMETRIC_KEY_LENGTH
ITERATIONS
